File: app/rcc/scripts/lua/fixed_width_fields.lua
--
-- Fixed-Width Fields Parser
--
-- Splits space-separated fixed-width fields into an array of values.
--
-- INPUT FORMAT: "John      25  Engineer  ABC123"
-- OUTPUT TABLE: {"John", "25", "Engineer", "ABC123"}
--

function parse(frame)
  local result = {}
  for field in frame:gmatch("%S+") do
    result[#result + 1] = field
  end
  return result
end

File: app/rcc/scripts/parser/js/fixed_width_fields.js
/**
 * Fixed-Width Fields Parser
 *
 * Parses space-separated fixed-width fields into an array of values.
 *
 * INPUT FORMAT: "John      25  Engineer  ABC123"
 * OUTPUT ARRAY: ["John", "25", "Engineer", "ABC123"]
 *
 * Note: Frame delimiters are automatically removed by Serial Studio.
 *       Multiple consecutive spaces are treated as field separators.
 */

//------------------------------------------------------------------------------
// Frame Parser Function
//------------------------------------------------------------------------------

/**
 * Parses a fixed-width format frame by splitting on whitespace.
 *
 * How it works:
 * 1. Splits the frame on one or more space characters
 * 2. Filters out any empty strings
 * 3. Returns array of extracted fields
 *
 * Example:
 *   Input:  "John      25  Engineer  ABC123"
 *   Output: ["John", "25", "Engineer", "ABC123"]
 *
 * @param {string} frame - The fixed-width data from the data source
 * @returns {array} Array of string values
 */
function parse(frame) {
  return frame.split(/\s+/).filter(value => value.length > 0);
}
